packages/php-domain/src/PhpDomain/MethodNameService.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3Docs\PhpDomain\PhpDomain;
+
+use Psr\Log\LoggerInterface;
+use T3Docs\PhpDomain\Nodes\MethodNameNode;
+
+class MethodNameService
+{
+    /** https://regex101.com/r/QrrSXk/1 */
+    private const PATTERN = '/^\s*(\w+)\s*\(\s*(.*?)\s*\)\s*(?::\s*(\w+))?\s*$/';
+
+    public function __construct(
+        private readonly LoggerInterface $logger
+    ) {}
+
+    public function getMethodName(string $name): MethodNameNode
+    {
+        if (preg_match(self::PATTERN, $name, $matches)) {
+            $methodName = $matches[1];
+            $parameters = isset($matches[2]) ? $matches[2] : '';
+            $returnType = $matches[3] ?? null;
+            $parametersArray = preg_split('/\s*,\s*/', $parameters, -1, PREG_SPLIT_NO_EMPTY);
+            if ($parametersArray === false) {
+                $parametersArray = [];
+            }
+            return new MethodNameNode($methodName, $parametersArray, $returnType);
+        }
+        $this->logger->warning(sprintf('Method name %s is invalid. See https://regex101.com/r/QrrSXk/1 for which kind of names are allowed.', $name));
```

Comment:
```suggestion
        $this->logger->warning(sprintf('Method signature %s is invalid. See https://regex101.com/r/QrrSXk/1 for which kind of names are allowed.', $name));
```

As far as I understand the signature consists of the method name, the arguments and the return type. So this would be more appropriate.

Additionally, I suggest to remove the regex101 link in the warning: We allow all method signatures PHP processes. Also the link can change, when the regex examples change (mind the `/1` at the end of the URL which marks the version).

Reply:
done